torchvision/prototype/datasets/_builtin/README.md

```diff
@@ -72,35 +107,81 @@ def sha256sum(path, chunk_size=1024 * 1024):
+archives with regular suffixes (`.tar`, `.zip`, ...), the datapipe will contain
+tuples comprised of the path and the handle for every file in the archive.
+Otherwise the datapipe will only contain one of such tuples for the file
+specified by the resource.
+
+Since the datapipes are iterable in nature, some datapipes feature an in-memory
+buffer, e.g. `IterKeyZipper` and `Grouper`. There are two issues with that: 1.
+If not used carefully, this can easily overflow the host memory, since most
+datasets will not fit in completely. 2. This can lead to unnecessarily long
+warm-up times when data is buffered that is only needed at runtime.
+
+Thus, all buffered datapipes should be used as early as possible, e.g. zipping
+two datapipes of file handles rather than trying to zip already loaded images.
+
+There are two special datapipes that are not used through their class, but
+through the functions `hint_sharding` and `hint_shuffling`. As the name implies
+they only hint part in the datapipe graph where sharding and shuffling should
+take place, but are no-ops by default. They can be imported from
+`torchvision.prototype.datasets.utils._internal` and are required in each
+dataset.
+
+Finally, each item in the final datapipe should be a dictionary with `str` keys.
+There is no standardization of the names (yet!).
 
-Since the datapipes are iterable in nature, some datapipes feature an in-memory buffer, e.g. `IterKeyZipper` and `Grouper`. There are two issues with that:
-1. If not used carefully, this can easily overflow the host memory, since most datasets will not fit in completely.
-2. This can lead to unnecessarily long warm-up times when data is buffered that is only needed at runtime.
+## FAQ
 
-Thus, all buffered datapipes should be used as early as possible, e.g. zipping two datapipes of file handles rather than trying to zip already loaded images.
+### How do I start?
 
-There are two special datapipes that are not used through their class, but through the functions `hint_sharding` and `hint_shuffling`. As the name implies they only hint part in the datapipe graph where sharding and shuffling should take place, but are no-ops by default. They can be imported from `torchvision.prototype.datasets.utils._internal` and are required in each dataset.
+Get the skeleton of your dataset class ready with all 3 methods. For
+`_make_datapipe()`, you can just do `return resources_dp[0]` to get started.
+Then import the dataset class in
+`torchvision/prototype/datasets/_builtin/__init__.py`: this will automatically
+register the dataset and it will be instantiable via
+`datasets.load("mydataset")`. On a separate script, try something like
 
-Finally, each item in the final datapipe should be a dictionary with `str` keys. There is no standardization of the names (yet!).
+```py
+d = datasets.load("mydataset")
+for e in d:
+    print(e)  # this is the content of the datapipe returned by _make_datapipe()
```

Comment:
1. Especially for an example, let's be more explicit here
2. I usually add a `break` to avoid loading the whole datapipe every time and terminating the execution manually. No strong opinion on that.

```suggestion
from torchvision.prototype import datasets

dataset = datasets.load("mydataset")
for sample in dataset:
    # this is the content of an item in datapipe returned by _make_datapipe()
    print(sample)
    break
```